src/containers/dynamicarray.d

```diff
@@ -40,25 +62,28 @@ struct DynamicArray(T, bool supportGC = true)
 			import core.memory : GC;
 			GC.removeRange(arr.ptr);
 		}
-		Mallocator.instance.deallocate(arr);
+		allocator.deallocate(arr);
 	}
 
 	/// Slice operator overload
-	T[] opSlice() @nogc
+	auto opSlice(this This)() @nogc
 	{
-		return arr[0 .. l];
+		alias ET = ContainerElementType!(This, T);
+		return cast(ET[]) arr[0 .. l];
 	}
 
 	/// ditto
-	T[] opSlice(size_t a, size_t b) @nogc
+	auto opSlice(this This)(size_t a, size_t b) @nogc
 	{
-		return arr[a .. b];
+		alias ET = ContainerElementType!(This, T);
+		return cast(ET[]) arr[a .. b];
```

Comment:
This could be rewritten to use the other `opSlice` overload, removing all the redundant code.